src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,64 +113,67 @@ public NuGetProject DefaultNuGetProject
                 throw new ArgumentNullException(nameof(projectSystemFactory));
             }
 
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _serviceProvider = serviceProvider;
             _projectSystemCache = projectSystemCache;
             _projectSystemFactory = projectSystemFactory;
+            _logger = logger;
+        }
 
-            _dte = ServiceLocator.GetInstance<DTE>();
-            _vsSolution = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution>();
-            _vsMonitorSelection = ServiceLocator.GetGlobalService<SVsShellMonitorSelection, IVsMonitorSelection>();
+        public async Task InitializeAsync(IAsyncServiceProvider site)
+        {
```

Comment:
Sorry, didn't update the description yet. The idea behind this change is to move all threading work out of MEF as it imposes high risk of deadlocks in VS.